src/Sil/Component/Product/Model/ProductInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Product\Model;
+
+interface ProductInterface
+{
+    /**
+     * Retreive Option trought OptionLabel(s) as a associative array.
+     *
+     * e.g.: [
+     *      'optionLabel#1' => [
+     *          'option#1',
+     *          'option#2',
+     *      ]
+     *      'optionLabel#2' => [
+     *          'option#3',
+     *          'option#4',
+     *      ]
+     * ].
+     *
+     * @return array
+     */
+    public function getOptionsAsArray(): array;
```

Comment:
Is it really necessary in order to be compliant with a Product ?

Reply:
In fact this method is aimed to be used by a service wich will generate product variants based on this resulting array.

Reply:
This kind of conversion may be done in the service we need it.

Reply:
Yes. Moving this method to the service itself